vividus-extension-selenium/src/main/java/org/vividus/monitor/AbstractHarOnFailureMonitor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.monitor;
+
+import java.io.IOException;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+import com.browserup.harreader.model.Har;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+
+import org.jbehave.core.steps.NullStepMonitor;
+import org.vividus.reporter.event.AttachmentPublishEvent;
+import org.vividus.reporter.model.Attachment;
+import org.vividus.softassert.event.AssertionFailedEvent;
+import org.vividus.util.json.JsonUtils;
+
+public abstract class AbstractHarOnFailureMonitor extends NullStepMonitor
```

Comment:
I propose to reuse the logic of this class https://github.com/vividus-framework/vividus/blob/master/vividus-extension-selenium/src/main/java/org/vividus/monitor/AbstractScreenshotOnFailureMonitor.java